traveler_core_kit/src/main/java/com/guestlogix/travelercorekit/models/Question.java

```diff
@@ -16,13 +20,15 @@
     private String description;
     private List<ValidationRule> validationRules;
     private QuestionType type;
+    private SuggestedAnswer suggestedAnswer;
 
-    Question(@NonNull String id, @NonNull String title, String description, @NonNull QuestionType type, @NonNull List<ValidationRule> rules) {
+    Question(@NonNull String id, @NonNull String title, String description, @NonNull QuestionType type, @NonNull List<ValidationRule> rules, SuggestedAnswer suggestedAnswer) {
         this.id = id;
```

Comment:
All good dude. I would add an exception here and mark this constructor as throws so that the mapping layer can catch it and reject payloads that dont make sense.

basically the error case to handle here is you wanna throw an exception if the suggestedAnswer is not of the same type as the `QuestionType`....
Looking at this now I think there might be a better way to do this in Java to get full compiler help, but this runtime check (also compile time cause of the exception handling) will do as well :)

Reply:
cool idea. i can also try generics for that. ill try it out later when i had no task in hand ;)